src/Assert.php

```diff
@@ -1822,7 +1822,11 @@ public static function countBetween($array, $min, $max, $message = '')
      */
     public static function isList($array, $message = '')
     {
-        if (!\is_array($array) || $array !== \array_values($array)) {
+        if (
+            !\is_array($array)
+            || (function_exists('\array_is_list') && !\array_is_list($array))
```

Comment:
If `array_is_list` is exists and returns `true` then `$array !== \array_values($array)` will be checked. This is double check.